### ISSUE TITLE
cache per-model dependency in model archiver

### ISSUE THREAD

**Issue by hgong-snap** [pytorch/serve#1935]
### 🚀 The feature

Correct me if I'm wrong: Currently torchserve supports per-model dependency where user can specify `--requirements-file` when running `torch-model_archiver`. and when model gets loaded, torchserve creates some sort of python venv to install these extra packages. 

To reduce the wait time, is it possible to cache the dependency packages somehow so that when the model gets loaded, it can be ready for serving (almost) immediately without waiting for **several minutes** to install dependencies? 

### Motivation, pitch

see above

**Comment by hgong-snap** [comment]
we use following steps to setup per-model dependencies. LMK if the setup is wrong.

- for model related files, we build a package by specifying a `setup.py` and run `python setup.py bdist_wheel --universal` to generate a .whl file
- add `my_model_package.whl` in `requirements.in`
- provide `extra-files` and `requirements.in` when running `torch-model-archiver`
```
torch-model-archiver -f \
    --model-name=my_model \
    --version=1.0 \
    --handler=handler.py \
    --serialized-file=my_model.pt \
    --extra-files=my_model_package.whl \
    --requirements-file=requirements.in
```

**Comment by msaroufim** [comment]
So `requirements.txt` is as far as I can tell the way people recommend sharing pip environment settings but one workaround may be in `--extra-files=venv` so you actually package the venv with your your model and then you run in your handler `source activate venv`. I haven't tried this but conceptually it could work albeit it might be error prone

**Comment by hgong-snap** [comment]
thanks for the reply @msaroufim !

with that said, the handler will look like:

```
import os
os.system("source activate venv")

# now these packages can be imported from venv
import my_dependency
import my_model
```
?

**Comment by msaroufim** [comment]
Yeah that looks about right, I will caution you though in that I haven't seen people do this just because then the provenance of `venv` is funky, your model packaging will take longer 

Also I believe `os.system()` will spawn a seperate process so you want the activation to happen within the same python interpreter so maybe this helps https://stackoverflow.com/a/14792407

**Comment by hgong-snap** [comment]
> Yeah that looks about right, I will caution you though in that I haven't seen people do this just because then the provenance of `venv` is funky and your model packaging will take longer.
> 
> Also I believe `os.system()` will spawn a seperate process so you want the activation to happen within the same python interpreter so maybe this helps https://stackoverflow.com/a/14792407

thanks this is really helpful!  will try!

**Comment by hgong-snap** [comment]
actually, one question related to this:

suppose model specified `my_package` in requirements.in, but it is already installed in base image(aka. the requirement is already satisfied), will `my_package` re-installed for this model?  or will skip and use this package in base image ?
